app/models/host/discovered.rb

```diff
@@ -41,8 +41,12 @@ def self.import_host_and_facts facts
     # construct hostname
     prefix_from_settings = Setting[:discovery_prefix]
     hostname_prefix = prefix_from_settings if prefix_from_settings.present? && prefix_from_settings.match(/^[a-zA-Z].*/)
-    hostname_prefix ||= 'mac'
-    hostname = FacterUtils::bootif_mac(facts).try(:downcase).try(:gsub,/:/,'').try(:sub,/^/, hostname_prefix)
+
+    name_fact = return_first_valid_fact(Setting::Discovered.discovery_hostname_fact_array,facts)
+    raise(::Foreman::Exception.new(N_("Invalid facts: hash does not contain the required fact '%s'"), name_fact)) unless name_fact && name_fact.present?
+    hostname = normalize_string_for_hostname("#{hostname_prefix}#{name_fact}")
+    raise(::Foreman::Exception.new(N_("Invalid hostname: Could not compute a valid hostname for the discovered host"))) unless hostname && hostname.present?
+    Rails.logger.warn "Hostname does not start with an alphabetical character. While this is allowed by the relevant RFC, this might create issues with certian software" unless hostname.downcase.match /^[a-z]/
```

Comment:
Let's keep our log clean, just use "Hostname does not start with an alphabetical character" I guess and that's it :-)